### ISSUE TITLE
External compiler flags invalid bytecode for abstract contracts

### ISSUE THREAD

**Issue by shanecarey17** [trufflesuite/truffle#3180]
- [ ] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue

External compiler complains about abstract contracts with no bytecode.

## Steps to Reproduce

I have three config files, one top level and two which refer to projects in subdirectories p1 and p2. I omitted some parts of the config I don't believe are important.

truffle.js

```
compilers: {
        external: {
          command: 'bash truffle-compile.sh',
          targets: [{
              path: './artifacts/**/*.json'
          }]
        }
   }
```

truffle.p1.js

```
contracts_build_directory: 'artifacts/p1/',
compilers: {
    solc: {
        version: '0.4.22'
    }
  }
```

truffle.p2.js

```
contracts_build_directory: 'artifacts/p2/',
compilers: {
    solc: {
        version: '0.5.16'
    }
  }
```

truffle-compile.sh

```
npx truffle compile --config truffle.p1.js
npx truffle compile --config truffle.p2.js
```

## Expected Behavior

The subproject compilation tasks should run with their respective compiler versions and put output into `artifacts/`. Then the targets path should move all of those outputs to the top level build directory (separate from subproject build directories).

## Actual Results

There are interface classes in the subprojects which have a bytecode of `0x` in the outputted JSON. The contract schema flags these as invalid and does not move the contracts to the build folder. This validation either does not occur or does not fail with a direct call to `npx truffle compile --config truffle.p*.js`.

```
Error: Schema validation failed. Errors:

should match pattern "^0x0$|^0x([a-fA-F0-9]{2}|__.{38})+$" (pattern):
{
  dataPath: '.bytecode',
  schemaPath: '#/definitions/Bytecode/pattern',
  params: { pattern: '^0x0$|^0x([a-fA-F0-9]{2}|__.{38})+$' },
  **data: '0x',**
  parentSchema: { type: 'string', pattern: '^0x0$|^0x([a-fA-F0-9]{2}|__.{38})+$' }
}

    at Object.validate (/node_modules/truffle/build/webpack:/packages/contract-schema/index.js:265:1)
    at processTargets (/node_modules/truffle/build/webpack:/packages/external-compile/index.js:154:1)
    at processTarget (/node_modules/truffle/build/webpack:/packages/external-compile/index.js:210:1)
    at processTargets (/node_modules/truffle/build/webpack:/packages/external-compile/index.js:152:1)
    at /node_modules/truffle/build/webpack:/packages/external-compile/index.js:263:1
Truffle v5.1.35 (core: 5.1.35)
Node v12.16.3
```

## Environment

* Operating System: Mac OS X
* Ethereum client: ganache
* Truffle version (`truffle version`): v5.1.35
* node version (`node --version`): v12.16.3
* npm version (`npm --version`): 6.14.4

**Comment by gnidan** [comment]
Hm, good catch. I guess `0x` is a valid value for bytecode? I'll PR this shortly.

**Comment by eggplantzzz** [comment]
I think this issue may be resolved in Truffle 5.1.36, can you confirm?

**Comment by haltman-at** [comment]
Looks like this has been fixed, closing.  Let us know if it's not.
